zulipterminal/ui_tools/views.py

```diff
@@ -286,3 +292,64 @@ def keypress(self, size: Tuple[int, int], key: str) -> str:
             self.controller.narrow_to_user(UnreadPMButton(pm, email))
 
         return super(MiddleColumnView, self).keypress(size, key)
+
+
+class RightColumnView(urwid.Frame):
+    """
+    Displays the users list on the right side of the app.
+    """
+    def __init__(self, view: Any) -> None:
+        self.view = view
+        self.user_search = UserSearchBox(self)
+        urwid.connect_signal(self.user_search, 'change',
+                             self.update_user_list)
+        self.view.user_search = self.user_search
+        search_box = urwid.LineBox(self.user_search)
+        self.searching = False
+        super(RightColumnView, self).__init__(self.users_view(),
+                                              header=search_box)
+
+    @async
+    def update_user_list(self, search_box: Any, new_text: str) -> None:
+        if not self.view.controller.editor_mode:
+            return
+        # wait for any previously started search to finish to avoid
+        # displaying wrong user list.
+        while self.searching:
```

Comment:
@punchagan do you think there is a better way we can do this?

Reply:
I think using a [lock](https://docs.python.org/3/library/threading.html#lock-objects) would work well here.  Also, you could try using regex search and see if it is noticeably faster than what you have implemented.